swayfmt/tests/mod.rs

```diff
@@ -1,82 +1,95 @@
 use std::sync::Arc;
 use swayfmt::{config::user_def::FieldAlignment, Formatter};
 
-/// Checks if the formatter is producing the same output when given it's output.
-fn test_stability(formatted_input: String, formatter: Formatter) -> bool {
-    let mut formatter = formatter;
-    let formatted_sway_code =
-        Formatter::format(&mut formatter, Arc::from(formatted_input.clone()), None).unwrap();
-    formatted_input == formatted_sway_code
+#[macro_use]
+mod macros;
+
+/// Takes a configured formatter as input and formats a given input and checks the actual output against an
+/// expected output. There are two format passes to ensure that the received output does not change on a second pass.
+fn check_custom_config(unformatted: &str, expected: &str, formatter: &mut Formatter) {
+    let first_formatted = Formatter::format(formatter, Arc::from(unformatted), None).unwrap();
+    assert_eq_pretty!(first_formatted, expected);
+
+    let second_formatted =
+        Formatter::format(formatter, Arc::from(first_formatted.clone()), None).unwrap();
+    assert_eq_pretty!(second_formatted, first_formatted);
+}
+
+/// Using a default Formatter, formats a given input and checks the actual output against an expected
+/// output. There are two format passes to ensure that the received output does not change on a second pass.
+fn check(unformatted: &str, expected: &str) {
+    let mut formatter = Formatter::default();
+    let first_formatted = Formatter::format(&mut formatter, Arc::from(unformatted), None).unwrap();
+    assert_eq_pretty!(first_formatted, expected);
+
+    let second_formatted =
+        Formatter::format(&mut formatter, Arc::from(first_formatted.clone()), None).unwrap();
+    assert_eq_pretty!(second_formatted, first_formatted);
```

Comment:
You've just made a nice abstraction above that can simplify this a lot :)
```suggestion
    let mut formatter = Formatter::default();
    check_custom_config(unformatted, expected, &mut formatter);
```

Reply:
nice catch!